demomultistoreform/src/Controller/DemoMultistoreController.php

```diff
@@ -0,0 +1,237 @@
+        $contentBlockGrid = $contentBlockGridFactory->getGrid(new SearchCriteria());
+
+        // configuration form
+        $configurationForm = $this->get('prestashop.module.demo_multistore.content_block_configuration.form_handler')->getForm();
+
+        $contentBlocCount = $this->getDoctrine()
+            ->getRepository(ContentBlock::class)
+            ->count([]);
+
+        return $this->render('@Modules/demomultistoreform/views/templates/admin/index.html.twig', [
+            'title' => 'Content block list',
+            'contentBlockGrid' => $this->presentGrid($contentBlockGrid),
+            'configurationForm' => $configurationForm->createView(),
+            'help_link' => false,
+            'displayFixtureGeneratorLink' => $contentBlocCount === 0,
+        ]);
+    }
+
+    public function create(Request $request): Response
+    {
+        $formDataHandler = $this->get('prestashop.module.demo_multistore.form.identifiable_object.builder.content_block_form_builder');
+        $form = $formDataHandler->getForm();
+        $form->handleRequest($request);
+
+        $formHandler = $this->get('prestashop.module.demo_multistore.form.identifiable_object.handler.content_block_form_handler');
+        $result = $formHandler->handle($form);
+
+        if (null !== $result->getIdentifiableObjectId()) {
+            $this->addFlash(
+                'success',
+                $this->trans('Successful creation.', 'Admin.Notifications.Success')
+            );
+
+            return $this->redirectToRoute('demo_multistore');
+        }
+
+        return $this->render('@Modules/demomultistoreform/views/templates/admin/form.html.twig', [
+            'contentBlockForm' => $form->createView(),
+            'title' => 'Content block creation',
+            'help_link' => false,
+        ]);
+    }
+
+    public function edit(Request $request, int $contentBlockId): Response
+    {
+        $formBuilder = $this->get('prestashop.module.demo_multistore.form.identifiable_object.builder.content_block_form_builder');
+        $form = $formBuilder->getFormFor((int) $contentBlockId);
+        $form->handleRequest($request);
+
+        $formHandler = $this->get('prestashop.module.demo_multistore.form.identifiable_object.handler.content_block_form_handler');
+        $result = $formHandler->handleFor($contentBlockId, $form);
+
+        if (null !== $result->getIdentifiableObjectId()) {
+            $this->addFlash(
+                'success',
+                $this->trans('Successful edition.', 'Admin.Notifications.Success')
+            );
+
+            return $this->redirectToRoute('demo_multistore');
+        }
+
+        return $this->render('@Modules/demomultistoreform/views/templates/admin/form.html.twig', [
+            'contentBlockForm' => $form->createView(),
+            'title' => 'Content block edition',
+            'help_link' => false,
+        ]);
+    }
+
+    public function delete(Request $request, int $contentBlockId): Response
```

Comment:
@matthieu-rolland when we Delete content block from All shops it is not deleted from `PREFIX_content_block`